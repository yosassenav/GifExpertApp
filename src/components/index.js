export * from "../components/AddCategory";
export * from "../components/GifGrid";
export * from "../components/GifItem";
export * from "../components/DeleteButton";

