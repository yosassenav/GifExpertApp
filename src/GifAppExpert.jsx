import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = ()=>{

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = (newCategory) => {
        setCategories([...categories, 'test'])
    }


    const handleSubmit = (evt, string) => {
        evt.preventDefault();
        console.log("Event from child", evt)
        console.log(string)
        /* add logic here */
      
    };


    return(
    <>
    {/**Titulo  */}
        <h1>Gif Expert App!</h1>
        

    {/**Input */}
    <AddCategory handleSubmit={handleSubmit} />
    {/**Listado de Gif */}
    {/* <button onClick={onAddCategory}>Add Category</button> */}
        <ol>
            {categories.map((category)=>{
                return <li key={category}>{category}</li>})}
        </ol>
        {/**Gif Item */}
        
    </>);
}