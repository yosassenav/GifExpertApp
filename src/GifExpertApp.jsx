import { useState } from "react";
import { AddCategory, Gifgrid, DeleteButton } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);
  const [selectedCategory, setSelectedCategory] = useState('');

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  }

  const handleDeleteCategory = (categoryToDelete) => {
    if (!selectedCategory) {
      console.warn("No category selected!");
    }

    // Filter out the selected category
    const updatedCategories = categories.filter((category) => category !== categoryToDelete);

    // Update the state with the new list of categories
    setCategories(updatedCategories);

    // Reset the selected category
    setSelectedCategory('');
  }

  return (
    <>
      {/**Titulo */}
      <h1>Gif Expert App!</h1>

      {/**Input */}
      <AddCategory onNewCategories={(evt) => onAddCategory(evt)} />

      {/**Listado de Gif */}
      {categories.map((category) => (
        <div key={category}>
          <Gifgrid key={category} category={category} />
          <DeleteButton data-testid={`deleteCategory-${category}`} onClick={()=>handleDeleteCategory(category)} />
        </div>
      ))}

    </>
  );
};
