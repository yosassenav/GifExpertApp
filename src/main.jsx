import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifAppExpert'

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp/>
  </React.StrictMode>,
)
