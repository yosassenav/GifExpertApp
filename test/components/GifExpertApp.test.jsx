import { describe, test, expect} from '@jest/globals';
import { render } from '@testing-library/react';
import { GifExpertApp } from "../../src/GifExpertApp";

describe('Pruebas en GifExpertApp', () => {

    test('debe hacer match con el snapshot',()=>{
        const  {container} = render(<GifExpertApp/>);
        expect(container).toMatchSnapshot();

    });

});