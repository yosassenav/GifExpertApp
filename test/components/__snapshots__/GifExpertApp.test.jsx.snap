// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App!
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <div>
    <h2>
      Cargando...
    </h2>
    <h3>
      One Punch
    </h3>
    <div
      class="card-grid"
    />
    <button
      class="delete-button"
    >
      Delete Category
    </button>
  </div>
</div>
`;
